matchers.go

```diff
@@ -57,6 +57,11 @@ func HTTP1Fast(extMethods ...string) Matcher {
 	return PrefixMatcher(append(defaultHTTPMethods, extMethods...)...)
 }
 
+// TLS matches HTTPS requests.
+func TLS() Matcher {
```

Comment:
Perfect.

So you're suggest I look at the first three bytes instead of just the first, and verify the version matches as well.

Or do you think it's worth doing even more strict validation of the tls handshake?